src/wheelMenuModel.cpp

```diff
@@ -0,0 +1,135 @@
+#include "wheelMenuModel.h"
+
+#include "fileSaver.h"
+#include <QRandomGenerator>
+
+namespace model {
+
+WheelMenuModel::WheelMenuModel(QObject *parent)
+    : QAbstractListModel{parent}
+    , m_innerRowIndex(-1)
+    , m_saveThread(new QThread(this))
+    , m_saveWorker(new FileSaver())
+    , m_configSaveTimer(new QTimer(this))
+{
+    auto menuItemFactory = [](const QString &name, const QString &color) {
+        QList<ListItem> listItems;
+        int rand = QRandomGenerator::global()->bounded(3, 12);
+        for (int i = 0; i < rand; ++i) {
+            listItems.append(ListItem{color});
+        }
+        return MenuItem{name, color, listItems};
+    };
+
+    addMenuItem(std::move(menuItemFactory("name-red", "red")));
+    addMenuItem(std::move(menuItemFactory("name-orange", "orange")));
+    addMenuItem(std::move(menuItemFactory("name-yellow", "yellow")));
+    addMenuItem(std::move(menuItemFactory("name-green", "green")));
+    addMenuItem(std::move(menuItemFactory("name-cyan", "cyan")));
+    addMenuItem(std::move(menuItemFactory("name-blue", "blue")));
+    addMenuItem(std::move(menuItemFactory("name-violet", "violet")));
+
+    m_saveWorker->moveToThread(m_saveThread);
+    connect(this,
+            &WheelMenuModel::startSaveConfig,
+            m_saveWorker,
+            &FileSaver::saveConfig,
+            Qt::QueuedConnection);
+
+    connect(m_configSaveTimer, &QTimer::timeout, this, [this] {
+        std::pair<std::vector<MenuItemPtr> *, int> modelData{&m_menuItems, m_innerRowIndex};
+        m_saveWorker->copyData(modelData);
+        emit startSaveConfig();
+    });
+
+    m_saveThread->start();
+    m_configSaveTimer->start(5000);
+}
+
+WheelMenuModel::~WheelMenuModel()
+{
+    m_saveThread->quit();
```

Comment:
is some dtors missing here? or, probably, connections in ctor?

Reply:
yeah, my mistake. I need to add
```suggestion
    m_saveThread->quit();
    m_saveThread->wait();
```
and add before
    connect(m_saveThread, &QThread::finished, m_saveThread, &QThread::deleteLater);
    connect(m_saveThread, &QThread::finished, m_saveWorker, &QObject::deleteLater);

Reply:
:+1: